### PR TITLE
Forms refactor 1

### PR DESCRIPTION
**Pull request by MarekBodingerBA**
Commit by commit.

Commits are large, but they only do one thing at a time, the commit messages should explain everything, these are required changes to make the forms more functional, clean and testable in next iterations. The next steps are to have form states in providers and refactor uploading / scanning files when @skifahrer is done.

### CODE REVIEW

**Review by mpinter — APPROVED** [review]
:+1:

File: next/.eslintrc.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ module.exports = {
     '@typescript-eslint/no-shadow': ['error', { allow: ['event', 'value', 'key', 'error'] }],
     /** Too tedious to type every function return explicitly */
     '@typescript-eslint/explicit-function-return-type': 'off',
+    '@typescript-eslint/no-unused-vars': 'warn',
```
[the remark this answers is not in the document]
Comment (MarekBodingerBA) [reply next/.eslintrc.js:13]: Yea, I thought it's doing something else. You are right, reverted.

File: next/.eslintrc.js [hunk 2]
```diff
@@ -54,7 +55,7 @@ module.exports = {
     'lodash/prefer-noop': 'off',
     /** if comparing values in cx function or creating translations, it's overkill to create variables for that */
     'sonarjs/no-duplicate-string': 'warn',
-    'sonarjs/cognitive-complexity': 'warn',
+    'sonarjs/cognitive-complexity': 'off',
```
[the remark this answers is not in the document]
Comment (MarekBodingerBA) [reply next/.eslintrc.js:58]: I would leave this as off, we ALWAYS comment it out and are annoyed by it, and in 99% it's really not possible to reduce the complexity in that particular case.

File: next/components/forms/FormStateProvider.tsx [hunk 3]
```diff
@@ -12,46 +10,94 @@ import {
...
-  const { schema } = eform
+const FormStateContext = createContext<FormState | undefined>(undefined)
+
+interface FormStateProviderProps {
+  eformSlug: string
+  formDefinition: FormDefinition
+  initialFormData: InitialFormData
+}
+
+// TODO figure out if we need to step over uiSchemas, or having a single one is enough (seems like it is for now)
```
[the remark this answers is not in the document]
Comment (MarekBodingerBA) [reply next/components/forms/FormStateProvider.tsx:71]: Cool, removed.

**Pull request merged**